### ISSUE TITLE
L2 untargeted attack not working?! 

### ISSUE THREAD

**Issue by as791** [carlini/nn_robust_attacks#42]
@carlini I am trying to run your source code. But when I go for cifar10 untargeted settings I get all zero loss always and thus class is the same as the original prediction. Thus attack is failing. I have tried using the confidence of 0, 10, 20, 40 in all case it failed. Although for confidence > 0 the losses are not zero but still the attack fails as the class is not changed.
Whereas targeted attack works fine in all cases.
Can you give some hint about what might be the problem? I am using pre-softmax layer only for prediction, as used in the code. 

Also, to train the model used in your paper for cifar10 you have written decay of 0.5 in momentum, how to do that I have done the step decay of 0.5 in learning rate at 10 epochs rate and got 79.5% test acc.

**Comment by as791** [comment]
I have also tried VGG16 Pre-trained network for ImageNet dataset with pre-softmax layer in this case also untargeted setting is not working.

**Comment by carlini** [comment]
Just noticing this now. Did you figure it out? If you wouldn't mind, could you say what the issue was in case it's something that I can help others debug.

**Comment by as791** [comment]
Yeah. Actually, it was problem from my side, I forgot to initialise false to targeted variable while initialising the class object, so it was giving 0 as expected, sorry for this issue. 

Although, if you could tell about CIFAR10 model Trianing settings used in paper that would be nice. Or what I took is fine.
